cmd/data-scanner-metric.go

```diff
@@ -249,37 +249,69 @@ func (p *scannerMetrics) lastMinuteActions(a lifecycle.Action) AccElem {
 	return val
 }
 
-// setCycle updates the current cycle metrics.
-func (p *scannerMetrics) setCycle(c *currentScannerCycle) {
-	if c != nil {
-		c2 := c.clone()
-		c = &c2
-	}
+// addCycleEvent updates the current cycle metrics.
+func (p *scannerMetrics) bucketScanStarted(bucket string, cycle uint32) {
 	p.cycleInfoMu.Lock()
-	p.cycleInfo = c
-	p.cycleInfoMu.Unlock()
+	defer p.cycleInfoMu.Unlock()
+
+	info := p.cycleInfo[bucket]
+	info.current = uint64(cycle)
+	info.started = time.Now()
+	info.ongoing = true
+
+	p.cycleInfo[bucket] = info
+}
+
+func (p *scannerMetrics) bucketScanFinished(bucket string) {
+	p.cycleInfoMu.Lock()
+	defer p.cycleInfoMu.Unlock()
+
+	info := p.cycleInfo[bucket]
+	info.ongoing = false
+	info.cycleCompleted = append(info.cycleCompleted, time.Now())
+	if len(info.cycleCompleted) > 10 {
+		info.cycleCompleted = info.cycleCompleted[len(info.cycleCompleted)-10:]
+	}
+	p.cycleInfo[bucket] = info
+}
+
+type currentScannerCycle struct {
+	current        uint64
+	ongoing        bool
+	started        time.Time
+	cycleCompleted []time.Time
```

Comment:
suggestion: lets add a comment that last 10 cycle completed timestamps are maintained